### DISCUSSION TITLE
Help with Implementing Validation for Custom Plugin Form Components

### DISCUSSION THREAD

**Discussion by chungchunwang** [Budibase/budibase#10582] [Contributing]
Hi! I work on the [File Upload Plugin](https://github.com/chungchunwang/Budibase-File-Upload), and I am trying to add validation as per this [issue](https://github.com/chungchunwang/Budibase-File-Upload/issues/5). I have gotten validation working by referencing the [Field component](https://github.com/Budibase/budibase/blob/develop/packages/client/src/components/app/forms/Field.svelte), but currently the error text does not show up when the form is submitted, only after the value is saved using the fieldApi. Do you know how I could solve this issue? Here is the relevant code: https://github.com/chungchunwang/Budibase-File-Upload/blob/master/src/Component.svelte

**Comment by melohagan** [comment]
Hey @chungchunwang 

It may be because you're not making use of `fieldApi.setValue`.

Looking at the `InnerForm` code I found this:

```js
// Skip if the value is the same
if (!skipCheck && fieldState.value === value) {
  return true
}
```

Perhaps the validation is being skipped in your case - this is just a guess though. You could try [Setting the field value](https://docs.budibase.com/docs/custom-component#setting-the-field-value) to see if that helps.

**Reply by chungchunwang** [reply]
Thank you! I'll take a look.
